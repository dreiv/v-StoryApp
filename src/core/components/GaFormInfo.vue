<script setup lang="ts">
defineProps<{ label?: string }>()
</script>

<template>
  <div :class="$style.container">
    <div :class="$style.iconWrapper">
      <slot name="icon" />
    </div>

    <slot>{{ label }}</slot>
  </div>
</template>

<style module>
.container {
  display: flex;
  align-items: center;
  gap: var(--ga-size-spacing-03);

  font-size: var(--ga-text-xs-font-size);
  line-height: var(--ga-text-xs-line-height);
  font-family: var(--ga-font-family-primary);
  letter-spacing: var(--ga-text-xs-tracking);
}

.iconWrapper {
  display: flex;
  flex-shrink: 0;
  justify-content: center;
  align-items: center;
  align-self: start;
  box-sizing: content-box;

  margin: var(--ga-size-spacing-01) 0 var(--ga-size-spacing-01) var(--ga-size-spacing-01);
  width: var(--ga-size-spacing-05);
  height: var(--ga-size-spacing-05);

  overflow: hidden;
}
</style>
