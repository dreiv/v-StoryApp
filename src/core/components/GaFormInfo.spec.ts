import { shallowMount, type VueWrapper } from '@vue/test-utils'
import GaFormInfo from './GaFormInfo.vue'

describe('GaFormInfo', () => {
  let wrapper: VueWrapper

  beforeEach(() => {
    wrapper = shallowMount(GaFormInfo)
  })

  it('should render correctly with default props', () => {
    expect(wrapper.html()).toMatchSnapshot()
  })

  it('should render correctly with label prop', () => {
    wrapper = shallowMount(GaFormInfo, { props: { label: 'Test Label' } })
    expect(wrapper.html()).toMatchSnapshot()
    expect(wrapper.text()).toContain('Test Label')
  })

  it('should render correctly with default slot', () => {
    wrapper = shallowMount(GaFormInfo, { slots: { default: 'Default Slot Content' } })
    expect(wrapper.html()).toMatchSnapshot()
    expect(wrapper.text()).toContain('Default Slot Content')
  })

  it('should render correctly with icon slot', () => {
    wrapper = shallowMount(GaFormInfo, { slots: { icon: '<svg></svg>' } })
    expect(wrapper.html()).toMatchSnapshot()
    expect(wrapper.find('svg').exists()).toBe(true)
  })
})
