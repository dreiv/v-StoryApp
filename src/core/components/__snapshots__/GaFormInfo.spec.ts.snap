// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`GaFormInfo > should render correctly with default props 1`] = `
"<div class="container">
  <div class="iconWrapper"></div>
</div>"
`;

exports[`GaFormInfo > should render correctly with default slot 1`] = `
"<div class="container">
  <div class="iconWrapper"></div>Default Slot Content
</div>"
`;

exports[`GaFormInfo > should render correctly with icon slot 1`] = `
"<div class="container">
  <div class="iconWrapper"><svg></svg></div>
</div>"
`;

exports[`GaFormInfo > should render correctly with label prop 1`] = `
"<div class="container">
  <div class="iconWrapper"></div>Test Label
</div>"
`;
